test/common/router/router_test.cc

```diff
@@ -3551,6 +3558,92 @@ TEST_F(RouterTest, RetryTimeoutDuringRetryDelay) {
   EXPECT_TRUE(verifyHostUpstreamStats(0, 1));
 }
 
+TEST_F(RouterTest, MaxStreamDurationValidlyConfigured) {
+  NiceMock<Http::MockRequestEncoder> encoder1;
+  Http::ResponseDecoder* response_decoder = nullptr;
+  ON_CALL(cm_.conn_pool_.host_->cluster_, commonHttpProtocolOptions()).WillByDefault(Invoke([&]() {
+    auto common_http_protocol_options = envoy::config::core::v3::HttpProtocolOptions();
+    common_http_protocol_options.mutable_max_stream_duration()->MergeFrom(
+        ProtobufUtil::TimeUtil::MillisecondsToDuration(500));
+    return common_http_protocol_options;
```

Comment:
Can you set a default action in the mock which returns a default HttpProtocolOptions by member reference? Then your test can just modify the portion of the proto that you want.

edit: looks like you already have the default. All you need to do is modify the mock member. Same elsewhere.